src/node_i18n.cc

```diff
@@ -62,6 +72,300 @@ bool flag_icu_data_dir = false;
 
 namespace i18n {
 
+const size_t kStorageSize = 1024;
+
+Local<Value> ICUException(Environment* env,
+                          UErrorCode status,
+                          const char* msg) {
+  if (msg == nullptr || msg[0] == '\0')
+    msg = "Unspecified ICU Exception";
+  char buf[kStorageSize];
+  int len = snprintf(buf, sizeof(buf), "%s [%s]", msg, u_errorName(status));
+
+  Local<Value> e =
+      Exception::Error(String::NewFromUtf8(env->isolate(),
+                       buf, v8::NewStringType::kNormal,
+                       len).ToLocalChecked());
+  if (e.IsEmpty())
+    return Local<Value>();
```

Comment:
if you're returning empty handles as an indicator it's probably be more "V8-ish" have the return signature as a `MaybeLocal<Value>` instead. been trying to do that in other locations myself.